### ISSUE TITLE
Power '^' operator in SMT2 not supported?

### ISSUE THREAD

**Issue by benjholla** [SRI-CSL/yices2#133]
The Yices manual page 38 shows a `^` power operator, but when I try to run the following `yices-smt2 test.smt2` I get an error `(error "at line 2, column 13: undefined term: ^")`.  I have compiled Yices with latest master versions of yices2 and libpoly using the `--enable-mcsat` flag.

test2.smt:
```
(set-logic QF_NRA)
(assert (< (^ 2 2) 4.1))
(check-sat)
(exit)
```

Am I doing anything obviously wrong?

**Comment by ianamason** [comment]
Mixing languages? 
In the yices2 language (which is what the manual is describing) `test.ys` is:
```
(assert (< (^ 2 2) 4.1))
(check)
```
and
```
yices_main test.ys
sat
```
There are executables for each language. `yices_main` for  the yices language, `yices_smt2` for 
`smt2` etc

**Comment by ianamason** [comment]
Sorry did not mean to close this on you.

**Comment by benjholla** [comment]
I think you are right, I found the smt2 does not support the ^ operator on reals. From [https://smtlib.github.io/jSMTLIB/SMTLIBTutorial.pdf](https://smtlib.github.io/jSMTLIB/SMTLIBTutorial.pdf) I read that `the functions + - * / < > <= >= are defined for Real arguments`. Does this mean that I cannot perform power operations using the SMT2 interface?

**Comment by ianamason** [comment]
Looking [here](http://smtlib.cs.uiowa.edu/theories.shtml) confirms your suspicions.

Stick to the yices language. Much nicer.

**Comment by benjholla** [comment]
Sorry I am reopening this issue because I am still somewhat confused on what is supported where. I have compiled yices `--enable-mcsat`. Sorry again if this is totally obvious what I am doing wrong, I am just really scratching my head on this. I want to perform non-linear arithmetic using the power operator.

In the SMT2 frontend I can do NRA by modifying the example in the ReadMe:

`$ yices-smt2 test.ys`
```
(set-logic QF_NRA)
(declare-fun x () Real)
(declare-fun y () Real)
(assert (> (* (- x y) (- x y)) 10))
(check-sat)
(get-model)
```
returns 
```
sat
(= x 0.894427)
(= y 0.447214)
```

However as we discussed previously in this thread I cannot use the power operator in SMT2.

`$ yices-smt2 test.ys` 
```
(set-logic QF_NRA)
(declare-fun x () Real)
(declare-fun y () Real)
(assert (> (^ (- x y) 2) 10))
(check-sat)
(get-model)
```
returns
`(error "at line 8, column 13: undefined term: ^")`

If I use the yices2 language instead of SMT2 I get an error about not allowing NRA in both equivalent examples.

`$ yices test.ys`
```
(define x::real)
(define y::real)
(assert (> (* (- x y) (- x y)) 10))
(check)
```
returns
`non-linear arithmetic not supported (line 3, column 36)`

`$ yices test.ys`
```
(define x::real)
(define y::real)
(assert (> (^ (- x y) 2) 10))
(check)
```
returns
`non-linear arithmetic not supported (line 3, column 30)`

Although....
`$ yices test.ys`
```
(define x::real)
(define y::real)
(assert (< (^ 2 2) 4.1))
(check)
```
returns
`sat`

It seems I cannot involve any variables in the arithmetic. Do I need to somehow set the logic in the yices language?
